### PR TITLE
Update clang to 10.0

### PR DESCRIPTION
**Pull request by klecki**
Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

#### Why we need this PR?
Update to newer clang in deps to use the newest clang that's works with the build image.
11.0 requires newer libc.

#### What happened in this PR? 
 - What solution was applied:
     Update deps
 - Affected modules and functionalities:
     Docker
 - Key points relevant for the review:
      
 - Validation and testing:
     CI
 - Documentation (including examples):
     NA

**JIRA TASK**: *[Use DALI-XXXX or NA]*

### CODE REVIEW

**Comment by awolant** [comment]
Any reason for removing explicit version variable from the command?

**Comment by klecki** [comment]
> Any reason for removing explicit version variable from the command?

It's based on specific url, I tried using clang 11 and it also changed the SLES version (it didn't work, so I went back to 10.0), so there is more than one version that could be "configured" for this url. That's why I removed it.

**Review by awolant — APPROVED** [review]

File: dali/operators/image/resize/resize_attr.cc [hunk 1]
```diff
@@ -216,7 +216,7 @@ void ResizeAttr::PrepareResizeParams(const OpSpec &spec, const ArgumentWorkspace
     GetShapeLikeArgument<float>(size_arg_, spec, "size", ws, spatial_ndim_, N);
   }
 
-  max_size_.resize(spatial_ndim_, std::numeric_limits<int>::max());
+  max_size_.resize(spatial_ndim_, static_cast<float>(std::numeric_limits<int>::max()));
```
Comment (mzient) [inline dali/operators/image/resize/resize_attr.cc:219]:
```suggestion
  max_size_.resize(spatial_ndim_, std::numeric_limits<float>::max());
```

File: dali/operators/image/color/brightness_contrast.h [hunk 2]
```diff
@@ -34,7 +34,7 @@ namespace detail {
 template <typename T>
 constexpr float FullRange() {
   return std::is_integral<T>::value
-    ? std::numeric_limits<T>::max()
+    ? static_cast<float>(std::numeric_limits<T>::max())
```
Comment (mzient) [inline dali/operators/image/color/brightness_contrast.h:37]: phhh.... rly?

**Review by JanuszL — APPROVED** [review]

File: dali/operators/image/resize/resize_attr.cc (hunk 1 above)
Comment (klecki) [reply dali/operators/image/resize/resize_attr.cc]: done

File: dali/operators/image/color/brightness_contrast.h (hunk 2 above)
Comment (klecki) [reply dali/operators/image/color/brightness_contrast.h:37]: To make the compiler happy.

**Review by JanuszL — APPROVED** [review]

**Pull request merged**